### PR TITLE
Adding retry description into ASQ Transport

### PR DESCRIPTION
**Pull request by WojcikMike**
@Particular/azure-storage-queues-maintainers please review

### CODE REVIEW

File: docs/retry-for-asq.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+## When using multiple storage accounts
+This section contains analysis of multiple storage account support when doing retry.
+
+### Each endpoint resides in separate storage account
+ * If every account has it's own error queue (that is situated in corresponding storage accounts)
+  * ServiceControl need to have two instances each connected to 1 storage account
+  * Failed messages can be retried as the failed queue is in the same storage account as destination queue
+  * Each instance of SC can see only part of the conversation
+
```
Comment (SeanFeldman) [inline docs/retry-for-asq.md]: extra break line?

File: docs/retry-for-asq.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * Messages that are failed residing in error queue contains in `FailedQ` header name of the queue of the receiver
+ * ServiceControl while doing retry is calling send method with destination set to queue used in `FailedQ` header
+ * everything works as expected
+
+## When using multiple storage accounts
+This section contains analysis of multiple storage account support when doing retry.
+
+### Each endpoint resides in separate storage account
+ * If every account has it's own error queue (that is situated in corresponding storage accounts)
+  * ServiceControl need to have two instances each connected to 1 storage account
```
Comment (SeanFeldman) [inline docs/retry-for-asq.md]: > ServiceControl need to have two instances each connected to 1 storage account

SC doesn't need that, we need that :)
How about "An instance per storage account will be required"?

File: docs/retry-for-asq.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  * Failed messages can be retried as the failed queue is in the same storage account as destination queue
```
Comment (SeanFeldman) [inline docs/retry-for-asq.md]: > failed queue

"queue where it failed"?

File: docs/retry-for-asq.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  * Each instance of SC can see only part of the conversation
+  
+ * If there is one error queue on different storage account
```
Comment (SeanFeldman) [inline docs/retry-for-asq.md]: Not clear. If error queue is associated with an account SC is not configured with, then SC plainly cannot access error queue and there will be no way to retry failed messages from another account.

File: docs/retry-for-asq.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  * ServiceControl need to have only 1 instance connected to storage account that has error queue
+  * Failed messages can not be retried at this point of time, as SC would pass the value of FailedQ to Send method as destination. Operation would fail. As the SC uses v6 of ASQ transport in the end it calls:
+https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/6.2.1/src/Transport/AzureMessageQueueSender.cs#L41
+
+NOTE: For SC to work FaileQ field need to contains connection string after @. As the v6 of the AzureMessageQueueSender should work in that configuration properly.
```
Comment (SeanFeldman) [inline docs/retry-for-asq.md]: > For SC to work FaileQ 

Should be "FailedQ"

Comment (SeanFeldman) [inline docs/retry-for-asq.md]: Can you confirm this? I was under impression the FailedQ contains the queue name _only_.

Comment (SeanFeldman) [inline docs/retry-for-asq.md]: Perhaps to work it should contain `queuename@accountname`, not `queuename@connectionString` :)
I.e. support multiple accounts.

**Comment by SeanFeldman** [comment]
Added a few comments, :+1:

File: docs/retry-for-asq.md (hunk 4 above)
Comment (WojcikMike) [inline docs/retry-for-asq.md]: My intention was that if failed queue is on a different storage account than destination (I am not sure if this is even possible), and SC monitoring the storage account with failedQ.

**Comment by WojcikMike** [comment]
@SeanFeldman your comments addressed

**Comment by SimonCropp** [comment]
should this be targeting the release branch?

**Comment by SimonCropp** [comment]
@WojcikMike should this be in docs? cc @Particular/servicecontrol-maintainers @Particular/docs-maintainers

**Comment by johnsimons** [comment]
@SimonCropp I don't see why. This is doco for the ASQ Transport developers. I guess is just so that they have an understanding of the implications of changes have to SC.
There is nothing in the doco that is useful for a consumer of SC.

**Comment by SeanFeldman** [comment]
In a way this information is not just for ASQ, but for all of the transport.
This is an attempt to create a technical documentation for SC required by transports to evolve, which doesn't exist other than in peoples heads. It would be relevant for ASB (and other) transports as well.

**Comment by johnsimons** [comment]
I am happy to move it to https://github.com/Particular/ServiceControl/blob/master/docs/ ?

**Comment by WojcikMike** [comment]
Ok let me create a PR on SC repo then.

**Pull request closed without merging**